frontend/js/views/timeline.js

```diff
@@ -545,6 +546,34 @@ define([
                 html: true,
                 container: "body"
             });
+
+            // Maintain a stylesheet for structured annotations
+            function createCategoryStylesheet() {
+                var stylesheet = annotationTool.video.get("categories")
+                    .map(function (category) {
+                        var color = category.get("settings").color;
+                        return ".vis-item.category-" + category.id + "," +
+                            ".vis-item.vis-selected.category-" + category.id + "{" +
+                            "background-color:" + color + ";" +
+                            "color:" + (
+                                chroma(color).luminance() < 0.5
+                                    ? "white"
+                                    : "black"
+                            ) +
+                            ";}";
+                            }).join("");
+                return $("<style>" + stylesheet + "</style>")
```

Comment:
Why don't you use .insertRule instead of this?

Reply:
Because I didn't think of it, tbh. :wink: 

But also https://caniuse.com/#search=insertrule worries me a bit. Thoughts?